### DISCUSSION TITLE
regarding imageregion

### DISCUSSION THREAD

**Discussion by Snowyrs3** [Spawnova/ShinsImageScanClass#4] [Q&A] [answered]
Hey there! I had a quick question about image region, when you specify the h and w, can they be negative ? I know in your example in the YT video you used swords to look in the lower right of the landmark, but if I want to search the upper left of a landmark is that what I would do? use negative h and w ? 

Also is there a way to search multiple images in a OR configuration? :o

**Comment by Spawnova** [comment, marked as answer]
I've added support for negative value for width and height

If you needed a specific region though you would still want to add offsets based on found image, for example;

```AutoHotkey
if (scan.image("swords.png",0,x,y)) {
	invX := x + 17  ;inv x starts 17 pixels right of found image
	inyV := y + 25	;inv y starts 25 pixels down from found image
	invW := 177 ;from the new coordinates the width is 177
	invH := 253	;and the height 253  (roughly)
}

;or possibly lets use the bottom right corner (music icon)
if (scan.image("musicHarp.png",0,x,y)) {
	invX := x - 184  ;inv x starts 184 pixels left of found image
	inyV := y - 267	;inv y starts 267 pixels up from found image
	invW := 177 ;from the new coordinates the width is 177
	invH := 253	;and the height 253  (roughly)
}
```


Also, not sure what you mean with your second question, specifically what do you mean by OR configuration?

**Reply by Snowyrs3** [reply]
thank you so much! That works great! And I was thinking if there was a way to use multiple images in one search query, like this image or this image and it returns the location of the first one if found or if it didn't find it it'll try the next ones. But now that I think about it, I could just do separate searches and handle that within if then statements or something right? Like specifically I'm trying to find the image of bubbles for fishing but it's a very subtle animation and very similar to normal water texture so I want to add more images of bubbles.

**Reply by Spawnova** [reply]
Yeah for multiple images you would just do multiple searches, here is an example with comments that should make things easier to understand =P
```AutoHotkey
bubbles := ["bubbles1.png","bubbles2.png","bubbles3.png"] ;set up an array of images

scan := new ShinsImageScanClass("Runescape")
scan.AutoUpdate := 0 ;turn auto update off, so that we can scan a single frame for multiple images

settimer,checker,100
return

checker:
scan.Update() ;update the pixel buffer manually
for index,bubbleImage in bubbles ;loop through bubble array searching for images, then stop once one is found
{
	if (scan.Image(bubbleImage,0,x,y)) {
		msgbox % "Found match for " bubbleImage " at location: " x "," y ;do something here
		break ;break the loop after finding one
	}
}
return
```

**Reply by Snowyrs3** [reply]
Thank you so much! that is a very neat solution!! I ended up doing a bunch of else ifs haha. For loop is a great idea! Thank you so much!

**Reply by Spawnova** [reply]
No problem, also another user pointed out a typo in my above example, it should actually be 
```AutoHotkey
for index,bubbleImage in bubbles
```
I accidently swapped index and key there! whoops lol
I've made changes in the above post to fix it as well =P

**Reply by vavaGriphen** [reply]
msgbox % scan.Image("pic.png", 0, myX, MyY)
CoordMode, Mouse
MouseMove, myX, MyY 

have run the script look for pic location.  but the return X, Y, location is on  the upper left  pic.
is there a way to get the center location of the pic? thanks!

**Reply by Spawnova** [reply]
> have run the script look for pic location. but the return X, Y, location is on the upper left pic. is there a way to get the center location of the pic? thanks!

Sure, I've added a new parameter for Image() and ImageRegion(), they now center results by default, update your class file and it should work now  =P

**Reply by vavaGriphen** [reply]
Got It updated and working nicely. Thank so much for sharing.
one more question... how to High Light the object location were found ? Maybe a few flash outline be nice.
Thanks again!
